### DISCUSSION TITLE
What is the EOL strategy of future versions of docker moby

### DISCUSSION THREAD

**Discussion by AlenversFr** [moby/moby#45841] [Q&A] [answered]
Hello,

I'm opening this discussion because we desesperately need information about the lifecycle of Docker Moby.
Right now the only source of information is [https://endoflife.date/docker-engine](url) and it's not ideal.

Context :
Starting from docker 23.0 the semver strategy changes a lot compared to what we are used to.

At this point with the release of docker 23.0 in February and 24.0 in May I'd like to know what is the strategy regarding EOL for these 2 versions and if possible the next ones.

Impact :
It's quite challenging to upgrade a lot of VMs (800+) and we prefer to focus on LTS versions. We are used to patch minor versions but not so much with major.

As for now, my operation team is migrating in hurry docker 20.10 to 23.0 because of a misleading information indicating that 20.10 is EOL 1 month after 23.0 turns GA.

What will happen if there a 25.0 release in September (for exemple) ? 
Does docker 23.0 gonna be EOL ?

Thanks a lot

**Comment by cpuguy83** [comment, marked as answer]
We plan to add a support table to the repo.
The table would be akin to what is on [kernel.org](https://kernel.org/category/releases.html)
Example:

<img width="695" alt="image" src="https://github.com/moby/moby/assets/799078/3fa24162-6560-4670-890a-0bae840a9eef">

The idea here is we have maintainers who are willing to maintain backports for various releases.
Definitely happy to accept contributions from the community, but this still requires an active maintainer to sponsor that work.

My general advice here is if you want features stay on the latest release.
If you want stability stay on latest-1. We do make an effort to make sure releases are stabilized before we drop them.

We do not typically have breaking changes. If there is a breaking change it is either a bug (and should be fixed) or it is fixing a really bad problem.
In most cases such changes are gated by API version and you can continue to get the older behavior by using the older API version (which is specified in the HTTP request, and the `docker` CLI also can read the version from an env var).
We also typically provide additional options you can use to get the old behavior.
